### ISSUE TITLE
Using DataWarrior

### ISSUE THREAD

**Issue by drc007** [OpenSourceMalaria/OSM_To_Do_List#385]
DataWarrior (http://www.openmolecules.org/datawarrior/) is a free cheminformatics visualisation program. At the moment getting data into this program is a little tedious, but DataWarrior does have a scripting/macro interface. If anyone has any experience in using this facility it might be very useful if they could use it to provide a way to seamlessly import the latest OSM directly into DataWarrior.

**Comment by drc007** [comment]
If it helps I've written an iPython notebook to access the data http://www.macinchem.org/reviews/osm/osmipython.php  and a script for Vortex http://www.macinchem.org/reviews/vortex/tut26/scripting_vortex26.php

**Comment by MFernflower** [comment]
Seems like a solid bit of kit! I think I will use it to manage my massive base of hypotheticals

**Comment by neilfws** [comment]
By "seamlessly import", do you mean directly from the online Google spreadsheet?

I looked at DataWarrior and I don't think there is a way to import from online tabular data sources. The best I could do was export the Google document as TSV and open that from disk, which seemed to work well. I don't see how macros could improve that process.

**Comment by drc007** [comment]
Ideally it would be simply clicking a button or selecting a menu item like the Vortex script. I don't know enough about DataWarrior to know if this is possible.

**Comment by MFernflower** [comment]
Perhaps ask the datawarrior devs to add an import from google spreadsheet button? As datawarrior is made in java this should not be difficult - Anyone here know the devs well enough to request a feature?

**Comment by thsa** [comment]
Loading the data into DataWarrior currently it involves these steps:
- Open the following URL in your browser and save the file to your disk:
http://docs.google.com/spreadsheets/d/1Rvy6OiM291d1GN_cyT6eSw_C3lSuJ1jaR7AJa8hgGsc/export?format=tsv
- Rename the file extension to .txt and open the file from DataWarrior

For the upcoming version of DataWarrior, which is due in a few weeks, there will be a task to load data directly from any URL. Then, you can configure this task with above URL, put it into a small macro, and save the macro file in the DataWarrior/macros folder. In this case the DataWarrior menu would contain an item to directly retrieve and open the data from the GoogleDoc site.

**Comment by drc007** [comment]
@thsa Thanks for the news about the update, sounds perfect.

**Comment by mattodd** [comment]
This sounds very useful, yes, thanks for the info @thsa. Chris it'd be nice
to alert people to this possibility quickly on the 24th (#386) if there's
time. Sounds like the kind of thing you were describing.

On 14 May 2016 at 07:03, Chris Swain <notifications@github.com> wrote:

> @thsa <https://github.com/thsa> Thanks for the news about the update,
> sounds perfect.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenSourceMalaria/OSM_To_Do_List/issues/385#issuecomment-219158503>
>



-- 
MATTHEW TODD | Associate Professor
School of Chemistry | Faculty of Science

THE UNIVERSITY OF SYDNEY
Rm 519, F11 | The University of Sydney | NSW | 2006
T +61 2 9351 2180  | F +61 2 9351 3329  | M +61 415 274104
E matthew.todd@sydney.edu.au | W
http://sydney.edu.au/science/people/matthew.todd.php
W http://opensourcemalaria.org/ | W http://opensourcetb.org/ | W
http://opensourcepharma.net/

CRICOS 00026A
This email plus any attachments to it are confidential. Any unauthorised
use is strictly prohibited. If you receive this email in error, please
delete it and any attachments.

**Comment by drc007** [comment]
@mattodd I agree and it perfectly underlines process of asking a question in an open forum and getting an answer!

**Comment by MFernflower** [comment]
I had a play with datawarrior myself - It's quite a solid bit of kit and im looking forward to having the OSTB/OSM series in a single .dwar
